co-authors-plus.php

```diff
@@ -1753,9 +1753,11 @@ public function filter_pre_get_avatar_data_url( $args, $id ) {
 			return $args;
 		}
 		$coauthor = $this->get_coauthor_by( 'id', $id );
-		if ( false !== $coauthor && isset( $coauthor->type ) && 'guest-author' === $coauthor->type ) { 
+		if ( false !== $coauthor && isset( $coauthor->type ) && 'guest-author' === $coauthor->type ) {
 			if ( has_post_thumbnail( $id ) ) {
 				$args['url'] = get_the_post_thumbnail_url( $id, $this->gravatar_size );
+			} elseif ( isset( $coauthor->type ) && 'guest-author' === $coauthor->type ) {
```

Comment:
With L1756, this L1957 will be always true. This line might check `$coauthor->user_email` instead?

Reply:
Good catch. What was I thinking here? 🤭 Just fixed it as you're 100% correct. `isset( $coauthor->type ) && 'guest-author' === $coauthor->type` will always be true at this line.